File: Geometry/Q4. Rectangle Area/Q4_RectangleArea.py
class Solution(object):
    def computeArea(self, ax1, ay1, ax2, ay2, bx1, by1, bx2, by2):
        """
        :type ax1: int
        :type ay1: int
        :type ax2: int
        :type ay2: int
        :type bx1: int
        :type by1: int
        :type bx2: int
        :type by2: int
        :rtype: int
        """
        # Area of first rectangle
        areaA = (ax2 - ax1) * (ay2 - ay1)
        
        # Area of second rectangle
        areaB = (bx2 - bx1) * (by2 - by1)
        
        # Overlapping width
        overlapWidth = min(ax2, bx2) - max(ax1, bx1)
        
        # Overlapping height
        overlapHeight = min(ay2, by2) - max(ay1, by1)
        
        overlapArea = 0  # Initialize overlap area
        
        # Check if rectangles actually overlap
        if overlapWidth > 0 and overlapHeight > 0:
            overlapArea = overlapWidth * overlapHeight
        
        # Total area covered by both rectangles
        return areaA + areaB - overlapArea

File: Geometry/Q4. Rectangle Area/Q4_RectangleArea.java
class Solution {
    public int computeArea(int ax1, int ay1, int ax2, int ay2, int bx1, int by1, int bx2, int by2) {
        // Area of first rectangle
        int areaA = (ax2 - ax1) * (ay2 - ay1);
        
        // Area of second rectangle
        int areaB = (bx2 - bx1) * (by2 - by1);
        
        // Find overlap width (intersection in x direction)
        int overlapWidth = Math.min(ax2, bx2) - Math.max(ax1, bx1);
        
        // Find overlap height (intersection in y direction)
        int overlapHeight = Math.min(ay2, by2) - Math.max(ay1, by1);
        
        int overlapArea = 0; // Initialize overlap area
        
        // If rectangles actually overlap (width and height > 0)
        if (overlapWidth > 0 && overlapHeight > 0) {
            overlapArea = overlapWidth * overlapHeight;
        }
        
        // Total area = sum of both areas - overlapped area
        return areaA + areaB - overlapArea;
    }
}
